f.el

```diff
@@ -270,6 +270,23 @@ If APPEND is non-nil, append the DATA to the existing contents."
         (unless (f-directory? path)
           (f--destructive path (make-directory path)))))))
 
+(defun f-mkdir-recursive (&rest dirs)
+  "Recursively create each directory in DIRS."
+
+  (dolist (dir dirs)
```

Comment:
Turns out `f-mkdir` can actually create directories like `mkdir -p`
can. The following code works fine despite `/tmp/a` not initially
existing on my machine.

```emacs-lisp
(f-mkdir "/" "tmp" "a" "b" "c")
```

What `f-mkdir` cannot do is create a directory with a fully formed
path like `/tmp/a/b/c`. This changes my understanding of `f-mkdir`.

```emacs-lisp
(f-mkdir "/tmp/a/b/c") ;; does not work
(f-mkdir "/tmp/a") ;; works
```

This would make `f-mkdir-recursive` change its initial goal of being
able to create subdirectories to create subdirectories **from a fully
formed path**. It may be better to modify `f-mkdir` instead of
creating a new function.

Reply:
The fact it creates "~/a", "~/b", and "~/c" might be an unintended effect from the `f-expand` in the function’s definition, which is also the origin of #21. I think the original intent was to build the final path with each argument given to `f-mkdir`, but it turns out an absolute path among them will reset the final path.

Reply:
I see. Also, I was wondering about the logic behind `(f-split "~/a")` giving `("/" "~" "a" "b")` ? Otherwise my initial instinct was to use something like `(f-make (f-split path))` to get recursive behaviour from `f-make`. The "/" at the beginning makes it necessary to write more code around it - and in this case, I think a dedicated function might be a better/more user-friendly choice.

Reply:
The behavior of `(f-split "~/a")` was a bug which got fixed through a merge today, it now correctly gives `("~" "a" "b")`.

Reply:
Oooh ! That is awesome. I'll try and use it tomorrow, and if it is intuitive enough to use with `f-mkdir` for recursive calls, I think the use case for the function I am proposing here is non-existant.

Reply:
Something I just thought about is if `f-mkdir` receives a fully formed path, then it could be possible to split it with `f-split`, and then call `f-mkdir` with its result with something like `apply`. That way, one could call `f-mkdir` either its traditional way or with only one fully formed path.

Reply:
I was thinking of something like this splitting when I wrote above: `(f-mkdir (f-split path))`. But I think you are trying to say something even more general. Could you elaborate a bit more please?

Also, the newest `f-split` is not on MELPA yet. no?

Reply:
After thinking a bit about it, I think what I was proposing might be out of scope for `f-mkdir` itself. What I was thinking is it could be possible to give `f-mkdir` two possible behaviours:
- Either give it a path similar to what `f-split` might return
- Or give it one and only one full path that gets `f-split`ed 

But such a behaviour could be too confusing for `f.el` users, I think the documentation should be more clear as to what is permited as `f-mkdir`’s arguments. On the other hand, a new function could act as a helper for `f-mkdir`, something like this:

``` emacs-lisp
(defun f-mkdir-full-path (dir)
  "Create directory DIR.
This function is a helper to `f-mkdir' and accepts only one full
path."
  (f-mkdir (f-split dir)))
```

Reply:
Oh, and on the topic of MELPA, it updates regularly each day, once every two hours iirc. Melpa Stable however only updates packages when a new release is published, so Melpa Stable still distributes commit https://github.com/rejeep/f.el/commit/de6d4d40ddc844eee643e92d47b9d6a63fbebb48 which has the `v0.20.0` tag. I’m currently considering releasing `v0.21.0` once #21 is fixed and I see no new major issue being raised due to the recent commits.

Reply:
Yeah. The fact that `f-mkdir` can take multiple arguments is actually a bit confusing. For functions like `f-join` it is quite intuitive to expect this - because well, they join various inputs together. But for `f-mkdir` it only adds confusion for the user. 

It might be good to think this through in more depth. For instance, how would this work: `(f-mkdir "a" "b" "/c/d" "/e" "f")`? How many directories should it create? And more importantly:

1. Is the behaviour intuitive for the user? 
2. Does the documentation describe this behaviour clearly?

Once we have a good design/concept for `f-mkdir`, I think adding recursion (either to `f-mkdir` itself or as a separate function) should be pretty straightforward. I'm happy to think about this if you want, or if you have some ideas we can go with that.

I didn't know about #21. That looks like a very old (and to be honest, quite important) bug that needs squashing. Good luck!

Reply:
I’ve just pushed some commits fixing the behaviour of `f-join` and clarifying its behaviour as well as `f-mkdir`’s. Their usage should be much clearer now.